### ISSUE TITLE
Problems with DOMDocument::schemaValidate() 

### ISSUE THREAD

**Issue by kripper** [peachpiecompiler/peachpie#771]
Peachpie throws errors when validating XML schema.
I'm attaching a simple test case that works fine with PHP 7.1 (and other versions) and fails with Peachpie.
The attached test case 1) throws "Wrong XML version information" on Peachpie, but also throws 2) another error that suggests that maybe Peachpie's schemaValidate is not able to validate correctly against types defined in .xsd files referenced with `<xs:include/>` (located in the same path as the main .xsd file).
In case you manage to fix the first error (by improving Peachpie compatibility or changing the XML), you will probably see the second error.

[test-case.zip](https://github.com/peachpiecompiler/peachpie/files/4802948/test-case.zip)

**Comment by jakubmisek** [comment]
thanks, so expected behavior is no error, no warning, no validation issues?

whilst peachpie throws an exception, and also it does not report the exception as a notice, instead it kills the application?

**Comment by kripper** [comment]
Expected behavior is no errors and pass the validation. For our use case, it is ok if the behavior is not exactly the same as with Zend (warnings could be different), but we need to know if the xml validates against the given schema.
Zend returns "ok", while Peachpie throws validation errors and stops execution.

**Comment by jakubmisek** [comment]
peachpie throws error or exception? what do you mean by "stops execution"?

**Comment by kripper** [comment]
Peachpie throws an exception:
```
Unhandled exception. System.ArgumentException: Wrong XML version information. The XML must match production "VersionNum ::= '1.' [0-9]+".
   at System.Xml.XmlDeclaration..ctor(String version, String encoding, String standalone, XmlDocument doc)
   at System.Xml.XmlDocument.CreateXmlDeclaration(String version, String encoding, String standalone)
   at Peachpie.Library.XmlDom.DOMDocument.__construct(String version, String encoding)
   at Peachpie.Library.XmlDom.DOMDocument..ctor(String version, String encoding)
   at <Root>.program_php.<Main>(Context <ctx>, PhpArray <locals>, Object this, RuntimeTypeHandle <self>) in [...]\test-case\program.php:line 50
   at <Script>.Main(String[] args)
```

I added a .msbuildproj file to run the test-case with peachpie:
[test-case.zip](https://github.com/peachpiecompiler/peachpie/files/4805289/test-case.zip)

**Comment by kripper** [comment]
Wait...first error was just because I was doing `$xml = new DOMDocument($xmlStr);`, but `$xml->loadXML($xmlStr);` works fine.

Here is the second error (that suggests that maybe Peachpie's schemaValidate is not able to validate correctly against types defined in .xsd files referenced with <xs:include/> located in the same path as the main .xsd file).

I attached a new test case.
[test-case.zip](https://github.com/peachpiecompiler/peachpie/files/4805361/test-case.zip)

The exception is:
```
Unhandled exception. System.Xml.Schema.XmlSchemaValidationException: Type 'http://www.sii.cl/SiiDte:FolioRType' is not declared.
   at System.Xml.Schema.XmlSchemaValidator.SendValidationEvent(XmlSchemaValidationException e, XmlSeverityType severity)
   at System.Xml.Schema.XmlSchemaValidator.SendValidationEvent(XmlSchemaException e)
   at System.Xml.Schema.XmlSchemaValidator.RecompileSchemaSet()
   at System.Xml.Schema.XmlSchemaValidator.Init()
   at System.Xml.Schema.XmlSchemaValidator..ctor(XmlNameTable nameTable, XmlSchemaSet schemas, IXmlNamespaceResolver namespaceResolver, XmlSchemaValidationFlags validationFlags)
   at System.Xml.DocumentSchemaValidator.CreateValidator(XmlSchemaObject partialValidationType, XmlSchemaValidationFlags validationFlags)
   at System.Xml.DocumentSchemaValidator.Validate(XmlNode nodeToValidate)
   at System.Xml.XmlDocument.Validate(ValidationEventHandler validationEventHandler, XmlNode nodeToValidate)
   at System.Xml.XmlDocument.Validate(ValidationEventHandler validationEventHandler)
   at Peachpie.Library.XmlDom.DOMDocument.schemaValidate(Context ctx, String schemaFile, Int32 flags)
   at CallSite.Target(Closure , CallSite , PhpValue , Context , CallerTypeParam , String )
   at System.Dynamic.UpdateDelegates.UpdateAndExecuteVoid4[T0,T1,T2,T3](CallSite site, T0 arg0, T1 arg1, T2 arg2, T3 arg3)
   at <Root>.program_php.<Main>(Context <ctx>, PhpArray <locals>, Object this, RuntimeTypeHandle <self>) in [...]\test-case\program.php:line 54
   at <Script>.Main(String[] args)
```

**Comment by kripper** [comment]
"FolioRType" is defined in "SiiTypes_v10.xsd" which is referenced in "dte.xsd".
I believe Peachpie fails because it is not able to access the referenced schema "SiiTypes_v10.xsd".

**Comment by roberthusak** [comment]
@kripper Thanks for the test case, `xs:include` and other references should work now.

I'd rather keep the check for the XML version number format (the first argument of `DOMDocument::__construct`) so that it can identify a wrong usage of the class. PHP allows any value, leading to weird results, e.g. `saveXML()` returning
```xml
<?xml version='<?xml version="1.0" encoding="ISO-8859-1"?>...'?>
```

**Comment by kripper** [comment]
Sure. Please not that first reported error was a false alarm. See: https://github.com/peachpiecompiler/peachpie/issues/771#issuecomment-646696691
